api/stse_device_authentication.c

```diff
@@ -1,182 +1,216 @@
-		challenge_size, 		/* random number size in bytes */
-		signature); 			/* returned signature */
-
-	if (ret != STSE_OK)
-	{
-		return ret;
-	}
-
-	/* Verify The Signature of the Random Number*/
-	ret = stse_certificate_verify_signature(
-			&leaf_certificate,
-			challenge,
-			challenge_size,
-			signature, (signature_size >> 1),
-			&signature[signature_size >> 1], (signature_size >> 1));
-
-	if(ret != STSE_OK)
-	{
-		return STSE_API_INVALID_SIGNATURE;
-	}
-	return STSE_OK;
-}
+/*!
+ ******************************************************************************
+ * \file	stse_device_authentication.c
+ * \brief   STSAFE-Axxx Middleware Authentication API (sources)
+ * \author  STMicroelectronics - CS application team
+ *
+ ******************************************************************************
+ * \attention
+ *
+ * <h2><center>&copy; COPYRIGHT 2023 STMicroelectronics</center></h2>
+ *
+ * This software is licensed under terms that can be found in the LICENSE file in
+ * the root directory of this software component.
+ * If no LICENSE file comes with this software, it is provided AS-IS.
+ *
+ *****************************************************************************/
+
+/* Includes ------------------------------------------------------------------*/
+
+#include <api/stse_ecc.h>
+#include <api/stse_device_authentication.h>
+
+#include "certificate/stse_certificate.h"
+#include "certificate/stse_certificate_subparsing.h"
+#include "certificate/stse_certificate_crypto.h"
+
+#define CERTIFICATE_SIZE_OFFSET_BYTES 	2U
+#define CERTIFICATE_SIZE_LENGTH 		2U
+#define CERTIFICATE_OFFSET_BYTES		0U
+#define CERTIFICATE_DEVICE_ID_OFFSET	15U
+#define CERTIFICATE_DEVICE_ID_SIZE		11U
+
+/* Static functions declaration ----------------------------------------------*/
+
+/* Exported functions --------------------------------------------------------*/
+
+stse_ReturnCode_t stse_get_device_id(stse_Handler_t * pSTSE, PLAT_UI8 Certificate_zone, PLAT_UI8* device_id)
+{
+	volatile stse_ReturnCode_t ret = STSE_API_INVALID_PARAMETER;
+
+	/* - Check stsafe handler initialization */
+	if (pSTSE == NULL)
+	{
+		return( STSE_API_HANDLER_NOT_INITIALISED );
+	}
+
+	ret = stse_data_storage_read_data_zone(
+			pSTSE,							/* STSAFE handler */
+			Certificate_zone,				/* Certificate zone */
+			CERTIFICATE_DEVICE_ID_OFFSET,	/* X bytes offset */
+			device_id,						/* Returned certificate size */
+			CERTIFICATE_DEVICE_ID_SIZE,		/* Certificate size length */
+			0,								/* No maximum chunck size (No chunck at all) */
+			STSAFEA_NO_PROT);				/* No protection */
+
+	return ret;
+}
+
+stse_ReturnCode_t stse_get_device_certificate_size(stse_Handler_t * pSTSE, PLAT_UI8 Certificate_zone, PLAT_UI16 * certificate_size)
+{
+	volatile stse_ReturnCode_t ret = STSE_API_INVALID_PARAMETER;
+	PLAT_UI8 certificate_size_ui8[2];
+
+	/* - Check stsafe handler initialization */
+	if (pSTSE == NULL)
+	{
+		return( STSE_API_HANDLER_NOT_INITIALISED );
+	}
+

```

Comment:
- Add a check on certificate_size to avoid NULL pointer:
- Prepend `certificate_size` with `p` => `pCertificate_size`

```C
	if (pCertificate_size == NULL)
	{
		return( STSE_API_INVALID_PARAMETER );
	}
```